src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -1338,6 +1249,54 @@ internal static string GetModulePath()
             return currentModulePath;
         }
 
+#if !UNIX
+        /// <summary>
+        /// Returns a PSModulePath suiteable for Windows PowerShell by removing this PowerShell's specific
+        /// paths from current PSModulePath.
+        /// </summary>
+        /// <returns>
+        /// Returns appropriate PSModulePath for Windows PowerShell.
+        /// </returns>
+        internal static string GetWindowsPowerShellModulePath()
+        {
+            string currentModulePath = GetModulePath();
+
+            if (currentModulePath == null)
+            {
+                return null;
+            }
+
+            // Remove any PowerShell specific paths including if set in powershell.config.json file
+            string[] modulePaths = new string[] {
+                GetPersonalModulePath(),
+                GetSharedModulePath(),
+                GetPSHomeModulePath(),
+                PowerShellConfig.Instance.GetModulePath(ConfigScope.AllUsers),
+                PowerShellConfig.Instance.GetModulePath(ConfigScope.CurrentUser)
+            };
```

Comment:
Better to use a `HashSet` here instead, then we can remove the `currentModulePath.Replace` loop. In the `foreach (var path in currentModulePath.Split(';'))` loop, just check if the path contains in the set, and skip adding to `modulePathList` if so.